File: backend/app/models/order.py
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel

PaymentMethod = Literal["GCASH", "BANK_TRANSFER", "CASH_AT_EVENT"]
PaymentStatus = Literal["PENDING", "PAID", "FAILED", "CANCELLED"]
FulfillmentStatus = Literal["PENDING", "READY", "CLAIMED", "CANCELLED"]


class Order(BaseModel):
    """Mirrors a row in the Orders sheet."""

    order_id: str
    user_id: str
    order_date: datetime
    total_amount: float
    payment_method: PaymentMethod
    payment_status: PaymentStatus = "PENDING"
    fulfillment_status: FulfillmentStatus = "PENDING"
    notes: str = ""
    created_at: datetime
    updated_at: datetime
    claimed_at: Optional[datetime] = None
    claimed_by: str = ""


class OrderItem(BaseModel):
    """Mirrors a row in the Order_Items sheet.

    product_name and unit_price are captured at purchase time so historical
    orders stay accurate even if the product is later renamed or repriced.
    bundle_id is set when this line was generated from resolving a bundle
    into its component products; empty for items purchased directly.
    """

    order_item_id: str
    order_id: str
    product_id: str
    product_name: str
    quantity: int
    unit_price: float
    subtotal: float
    variant: str = ""
    bundle_id: str = ""

File: frontend/types/order.ts
import { CartLineInput } from "./cart";

export type PaymentMethod = "GCASH" | "BANK_TRANSFER" | "CASH_AT_EVENT";
export type PaymentStatus = "PENDING" | "PAID" | "FAILED" | "CANCELLED";
export type FulfillmentStatus = "PENDING" | "READY" | "CLAIMED" | "CANCELLED";

export interface OrderItem {
  order_item_id: string;
  product_id: string;
  product_name: string;
  quantity: number;
  unit_price: number;
  subtotal: number;
  variant: string;
  bundle_id: string;
}

export interface Order {
  order_id: string;
  user_id: string;
  parent_name: string;
  order_date: string;
  total_amount: number;
  payment_method: PaymentMethod;
  payment_status: PaymentStatus;
  fulfillment_status: FulfillmentStatus;
  notes: string;
  created_at: string;
  updated_at: string;
  claimed_at: string | null;
  claimed_by: string;
  items: OrderItem[];
}

export interface CheckoutRequest {
  items: CartLineInput[];
  parent_name: string;
  contact_number: string;
  payment_method: PaymentMethod;
  notes?: string;
  idempotency_key?: string;
}
